shank-render/src/pda/render_pda.rs

```diff
@@ -0,0 +1,305 @@
+use proc_macro2::{Span, TokenStream};
+use quote::quote;
+use shank_macro_impl::{
+    parsed_struct::{ProcessedSeed, Seed},
+    syn::Ident,
+};
+
+use crate::consts::solana_program_pubkey;
+
+use super::render_args_comments;
+
+pub fn render_pda_fn(
+    processed_seeds: &[ProcessedSeed],
+    seeds_fn_name: &Ident,
+    seeds_fn_with_bump_name: &Ident,
+    pda_fn_name: &Ident,
+    pda_fn_with_bump_name: &Ident,
+    include_comments: bool,
+) -> Option<TokenStream> {
+    let RenderedPdaParts {
+        seed_param_assigns,
+        seed_fn_args,
+        pda_fn_args,
+    } = render_pda_parts(processed_seeds);
+    if pda_fn_args.is_empty() {
+        return None;
+    }
+
+    let pubkey = solana_program_pubkey();
+    let seed_bump_arg = if seed_fn_args.is_empty() {
+        quote! { bump_arg }
+    } else {
+        quote! { , bump_arg }
+    };
+    let (pda_comments, pda_with_bump_comments) = if include_comments {
+        let args_comments = render_args_comments(processed_seeds, true);
+        (
+            format!(
+                r#"
+                /// Derives the PDA for this account.
+                ///
+                /// * **program_id**: The id of the program
+                {}"#,
+                args_comments.join("\n")
+            )
+            .parse()
+            .unwrap(),
+            format!(
+                r#"
+                /// Derives the PDA for this account allowing to provide a bump seed.
+                ///
+                /// * **program_id**: The id of the program
+                {}
+                /// * **bump**: the bump seed to pass when deriving the PDA"#,
+                args_comments.join("\n")
+            )
+            .parse()
+            .unwrap(),
```

Comment:
```suggestion
            .to_token_stream()
```
might be better than `parse().unwrap()` ?